src/main/java/org/coffeeprojects/pmda/feature/project/service/impl/RedmineProjectService.java

```diff
@@ -31,7 +30,12 @@ public void updateLastCheckProject(ProjectEntity projectEntity) {
     }
 
     @Transactional
-    public ProjectEntity initializeProject(TrackerParametersBean tracker) {
+    public void deactivateProject(TrackerParametersBean tracker) {
+        log.debug("Redmine - deactivate project");
+    }
+
+    @Transactional
+    public ProjectEntity initializeProject(TrackerParametersBean tracker, boolean hasDeactivated) {
         log.debug("Redmine - initialize project");
         return (ProjectEntity) new ProjectEntity().setId(new CompositeIdBaseEntity().setTrackerType(TrackerTypeEnum.REDMINE));
```

Comment:
Je pense que le cast est inutile (c'est déjà un projetEntity)

Reply:
Non, car setId retourne un BaseEntity, donc obligé de faire le cast

Reply:
Il faut que BaseEntity soit de type generique si tu veux eviter les casts 
class BaseEntity<T extends BaseEntity>

et apres 

public T setId(id) {
    ....
    return (T) this
}